frame/benchmarking/src/lib.rs

```diff
@@ -794,100 +794,115 @@ macro_rules! impl_benchmark {
 				// Default number of steps for a component.
 				let mut prev_steps = 10;
 
-				// Select the component we will be benchmarking. Each component will be benchmarked.
-				for (idx, (name, low, high)) in components.iter().enumerate() {
-					// Get the number of steps for this component.
-					let steps = steps.get(idx).cloned().unwrap_or(prev_steps);
-					prev_steps = steps;
-
-					// Skip this loop if steps is zero
-					if steps == 0 { continue }
-
-					let lowest = lowest_range_values.get(idx).cloned().unwrap_or(*low);
-					let highest = highest_range_values.get(idx).cloned().unwrap_or(*high);
-
-					let diff = highest - lowest;
+				let repeat_benchmark = |
```

Comment:
Git makes it look a lot has changed here, but what I did was basically:

* Take the stuff inside the "components loop", and turned it into a closure.
* If no components, directly execute this closure.
* if components, then loop over components while executing this closure in each loop.